### PR TITLE
Translate "Your First Component" to Telugu

### LINKED ISSUE #21: Your First Component
URL - https://te-react-dev.vercel.app/learn/your-first-component
Requires modification to [src/content/learn/your-first-component.md](https://github.com/reactjs/te.react.dev/tree/main/src/content/learn/your-first-component.md) this file.

If you want to translate this page, please comment below so I can assign you.

### PR DESCRIPTION
**Pull request by Shiva-Sai-ssb**
I have translated the `Your First Component` section into Telugu. 
I believe I have completed everything correctly, do let me know if any changes or improvements are required.

Closes #21

### CODE REVIEW

**Comment by Shiva-Sai-ssb** [comment]
Hi @srikanth-kandi, could you please review it when you have a moment?

**Comment by srikanth-kandi** [comment]
> Hi @srikanth-kandi, could you please review it when you have a moment?

Sure will review, currently I'm bit occupied.

File: src/content/learn/your-first-component.md [hunk 1]
```diff
@@ -203,44 +203,44 @@ function Profile() {
...
+మొత్తం React యాప్స్ కంపోనెంట్స్‌ను అన్ని దారిలో ఉపయోగిస్తాయి. అంటే, మీరు కేవలం రీయూజబుల్ భాగాలు, ఉదాహరణకు బటన్‌లతో మాత్రమే కంపోనెంట్స్‌ను ఉపయోగించరు, కానీ పెద్ద భాగాలు, ఉదాహరణకు సైడ్‌బార్లు, జాబితాలు, మరియు చివరగా, పూర్తి పేజీలు కూడా! కంపోనెంట్స్ UI కోడ్ మరియు మార్కప్‌ను వ్యవస్థీకరించడానికి చాలా ఉపయోగకరమైన మార్గం, కొన్నిటిని ఒకసారి మాత్రమే ఉపయోగించినా.
 
-[React-based frameworks](/learn/start-a-new-react-project) take this a step further. Instead of using an empty HTML file and letting React "take over" managing the page with JavaScript, they *also* generate the HTML automatically from your React components. This allows your app to show some content before the JavaScript code loads.
+[React ఆధారిత ఫ్రేమ్‌వర్క్‌లు](/learn/start-a-new-react-project) ఈ దశను మరింత ముందుకు తీసుకుంటాయి. ఒక ఖాళీ HTML ఫైల్‌ను ఉపయోగించి, React "పేజీని నిర్వహించేందుకు" జావాస్క్రిప్ట్‌తో పేజీని నిర్వహించడానికి అనుమతించే బదులుగా, అవి *మీ రియాక్ట్ కాంపొనెంట్ల నుండి HTMLని ఆటోమేటిగ్గా* జనరేట్ చేస్తాయి. ఇది జావాస్క్రిప్ట్ కోడ్ లోడ్ అవకముందు మీ యాప్ కొన్ని కంటెంట్‌ను చూపించడానికి అనుమతిస్తుంది.
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:218]:
```suggestion
[React ఆధారిత ఫ్రేమ్‌వర్క్‌లు](/learn/start-a-new-react-project) ఈ దశను మరింత ముందుకు తీసుకెళ్తుంది. ఖాళీ HTML ఫైల్ ఉపయోగించి React‌ తో JavaScript ద్వారా పేజీని "నియంత్రించేందుకు" అవకాశం ఇవ్వడం కంటే బదులు, React కంపోనెంట్ల నుండి HTML‌ ను ఆటోమేటిక్‌గా సృష్టించవచ్చు.  ఇది JavaScript కోడ్ లోడ్ అవ్వడానికి ముందే మీ యాప్ కొన్ని కంటెంట్‌ను ప్రదర్శించేందుకు అనుమతిస్తుంది.
```

File: src/content/learn/your-first-component.md [hunk 2]
```diff
@@ -203,44 +203,44 @@ function Profile() {
...
 
-Still, many websites only use React to [add interactivity to existing HTML pages.](/learn/add-react-to-an-existing-project#using-react-for-a-part-of-your-existing-page) They have many root components instead of a single one for the entire page. You can use as much—or as little—React as you need.
+అయితే, అనేక వెబ్‌సైట్లు Reactని కేవలం [మొదలైన HTML పేజీలకు ఇంటరాక్టివిటీ జోడించడానికి](/learn/add-react-to-an-existing-project#using-react-for-a-part-of-your-existing-page) మాత్రమే ఉపయోగిస్తాయి. అవి పేజీ మొత్తం కోసం ఒకే ఒక రూట్ కాంపొనెంట్ యొక్క స్థానంలో అనేక రూట్ కాంపొనెంట్లను కలిగి ఉంటాయి. మీరు ఎంత React ఉపయోగించాలనుకుంటే, అంతా లేదా కొంత React ఉపయోగించవచ్చు.
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:220]:
```suggestion
అయితే, అనేక వెబ్‌సైట్లు React ని కేవలం [కొన్ని HTML పేజీలకు ఇంటరాక్టివిటీ జోడించడానికి](/learn/add-react-to-an-existing-project#using-react-for-a-part-of-your-existing-page) మాత్రమే ఉపయోగిస్తాయి. అవి పేజీ మొత్తం కోసం ఒకే ఒక రూట్ కాంపొనెంట్ యొక్క స్థానంలో అనేక రూట్ కాంపొనెంట్లను కలిగి ఉంటాయి. మీరు ఎంత React ఉపయోగించాలనుకుంటే అంత లేదా కొంత React ఉపయోగించవచ్చు.
```

File: src/content/learn/your-first-component.md [hunk 3]
```diff
@@ -203,44 +203,44 @@ function Profile() {
...
 
 </DeepDive>
 
 <Recap>
 
-You've just gotten your first taste of React! Let's recap some key points.
+మీరు React ను మొదటి సారి అనుభవించారు! కొన్ని ముఖ్యమైన అంశాలను పునరావలంబం చేద్దాం.
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:226]:
```suggestion
మీరు React యొక్క మొదటి పరిచయాన్ని పొందారు! కొన్ని ముఖ్యమైన పాయింట్లను తిరిగి గుర్తు చేసుకుందాం.
```

File: src/content/learn/your-first-component.md [hunk 4]
```diff
@@ -203,44 +203,44 @@ function Profile() {
...
 
-* React lets you create components, **reusable UI elements for your app.**
-* In a React app, every piece of UI is a component.
-* React components are regular JavaScript functions except:
+* React మీకు **మీ యాప్ కోసం రీయూజబుల్ UI ఎలిమెంట్స్** అయిన కంపోనెంట్లను సృష్టించడానికి అనుమతిస్తుంది.
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:228]:
```suggestion
* React మీరు **మీ యాప్ కోసం రీయూజబుల్ UI ఎలిమెంట్స్** అయిన కంపోనెంట్లను సృష్టించడానికి అనుమతిస్తుంది.
```

File: src/content/learn/your-first-component.md [hunk 5]
```diff
@@ -203,44 +203,44 @@ function Profile() {
...
+* React యాప్‌లో ప్రతి UI భాగం ఒక కంపోనెంట్.
+* React కంపోనెంట్లు సాధారణ జావాస్క్రిప్ట్ ఫంక్షన్లు అయినప్పటికీ:
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:230]:
```suggestion
* React కంపోనెంట్లు సాధారణ JavaScript ఫంక్షన్లు అయినప్పటికీ:
```

File: src/content/learn/your-first-component.md [hunk 6]
```diff
@@ -203,44 +203,44 @@ function Profile() {
...
 
-  1. Their names always begin with a capital letter.
-  2. They return JSX markup.
+  1. వాటి పేర్లు ఎల్లప్పుడూ పెద్ద అక్షరంతో ప్రారంభం అవుతాయి.
+  2. అవి JSX మార్కప్‌ను return ఇస్తాయీ.
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:233]:
```suggestion
  2. అవి JSX మార్కప్‌ను return చేస్తాయి.
```

File: src/content/learn/your-first-component.md [hunk 7]
```diff
@@ -261,11 +261,11 @@ img { height: 181px; }
 
 </Sandpack>
 
-Try to fix it yourself before looking at the solution!
+పరిష్కారాన్ని చూడటానికి ముందు దానిని మీరు స్వయంగా సరిచేయడానికి ప్రయత్నించండి!
 
 <Solution>
 
-Add `export default` before the function definition like so:
+ఫంక్షన్ నిర్వచనానికి ముందు `export default` జోడించండి ఇలా:
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:268]:
```suggestion
ఫంక్షన్ డెఫినేషన్ కి ముందు `export default` ని జోడించండి ఇలా:
```

File: src/content/learn/your-first-component.md [hunk 8]
```diff
@@ -286,17 +286,17 @@ img { height: 181px; }
 
 </Sandpack>
 
-You might be wondering why writing `export` alone is not enough to fix this example. You can learn the difference between `export` and `export default` in [Importing and Exporting Components.](/learn/importing-and-exporting-components)
+మీరు ఈ ఉదాహరణను సరిచేయడానికి కేవలం `export` వ్రాయడం ఎందుకు సరిపోదో అనుకుంటున్నారో. `export` మరియు `export default` మధ్య తేడా తెలుసుకోడానికి [కంపోనెంట్లను ఇంపోర్ట్ చేసుకోవడం మరియు ఎక్స్‌పోర్ట్ చేసుకోవడం](/learn/importing-and-exporting-components) చూడండి.
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:289]:
```suggestion
మీరు ఈ ఉదాహరణను సరిచేయడానికి కేవలం `export` వ్రాయడం ఎందుకు సరిపోదు అని ఆలోచిస్తున్నట్లయితే. `export` మరియు `export default` మధ్య తేడా తెలుసుకోడానికి [కంపోనెంట్లను ఇంపోర్ట్ మరియు ఎక్స్‌పోర్ట్ చేసుకోవడం](/learn/importing-and-exporting-components) గురించి చూడండి.
```

File: src/content/learn/your-first-component.md [hunk 9]
```diff
@@ -286,17 +286,17 @@ img { height: 181px; }
...
 
 </Solution>
 
-#### Fix the return statement {/*fix-the-return-statement*/}
+#### రిటర్న్ స్టేట్‌మెంట్‌ని సరిచేయండి {/*fix-the-return-statement*/}
 
-Something isn't right about this `return` statement. Can you fix it?
+ఈ `return` స్టేట్‌మెంట్‌లో ఏదో సరైనది లేదు. మీరు దీన్ని సరిచేయగలరా?
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:295]:
```suggestion
ఈ `return` స్టేట్‌మెంట్‌లో ఏదో సరిగ్గా లేదు. మీరు దీన్ని సరిచేయగలరా?
```

File: src/content/learn/your-first-component.md [hunk 10]
```diff
@@ -286,17 +286,17 @@ img { height: 181px; }
 
 </Sandpack>
 
-You might be wondering why writing `export` alone is not enough to fix this example. You can learn the difference between `export` and `export default` in [Importing and Exporting Components.](/learn/importing-and-exporting-components)
+మీరు ఈ ఉదాహరణను సరిచేయడానికి కేవలం `export` వ్రాయడం ఎందుకు సరిపోదో అనుకుంటున్నారో. `export` మరియు `export default` మధ్య తేడా తెలుసుకోడానికి [కంపోనెంట్లను ఇంపోర్ట్ చేసుకోవడం మరియు ఎక్స్‌పోర్ట్ చేసుకోవడం](/learn/importing-and-exporting-components) చూడండి.
 
 </Solution>
 
-#### Fix the return statement {/*fix-the-return-statement*/}
+#### రిటర్న్ స్టేట్‌మెంట్‌ని సరిచేయండి {/*fix-the-return-statement*/}
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:293]:
```suggestion
#### return స్టేట్‌మెంట్‌ ని సరిచేయండి {/*fix-the-return-statement*/}
```

File: src/content/learn/your-first-component.md [hunk 11]
```diff
@@ -357,9 +357,9 @@ img { height: 180px; }
 
 </Solution>
 
-#### Spot the mistake {/*spot-the-mistake*/}
+#### పొరపాటును కనుగొనండి {/*spot-the-mistake*/}
 
-Something's wrong with how the `Profile` component is declared and used. Can you spot the mistake? (Try to remember how React distinguishes components from the regular HTML tags!)
+`Profile` కంపోనెంట్ ఎలా ప్రకటించబడిందో మరియు ఉపయోగించబడిందో ఒకసారి పరిశీలించండి. మీరు తప్పు ఎక్కడ ఉంది అన్నదాన్ని గుర్తించగలరా? (React ఎలా కంపోనెంట్లను సాధారణ HTML ట్యాగుల నుండి వేరు చేస్తుందో గుర్తు పెట్టుకోండి!)
```
Comment (srikanth-kandi) [inline src/content/learn/your-first-component.md:362]:
```suggestion
`Profile` కంపోనెంట్ ఎలా డిక్లేర్ చేయబడిందో మరియు ఉపయోగించబడిందో ఒకసారి పరిశీలించండి. మీరు తప్పు ఎక్కడ ఉంది అన్నదాన్ని గుర్తించగలరా? (React ఎలా కంపోనెంట్లను సాధారణ HTML ట్యాగుల నుండి వేరు చేస్తుందో గుర్తు తెచ్చుకోండి!)
```

**Review by srikanth-kandi — CHANGES_REQUESTED** [review]
Thank you @Shiva-Sai-ssb for your contribution!

Please apply the requested changes so I can approve your pull request.

**Comment by Shiva-Sai-ssb** [comment]
Hey @srikanth-kandi, I've made the changes you requested. Please check!

**Review by srikanth-kandi — APPROVED** [review]

**Pull request merged**